README.md

```diff
@@ -1,25 +1,109 @@
+
+## [LZ4](https://www.lz4.org/)
+LZ4 is an extremely fast compression algorithm that provides compression ratios comparable
+to Snappy and LZO. LZ4 is an excellent choice for applications that require high-performance
+compression and decompression.
+
+The native implementation of LZ4 is provided by the `Lz4NativeCompressor` and `Lz4NativeDecompressor` 
+classes. The Java implementation is provided by the `Lz4JavaCompressor` and `Lz4JavaDecompressor`
+
+## [Snappy](https://google.github.io/snappy/)
+Snappy is not as fast as LZ4, but provides a guarantee on memory usage that makes it a good
+choice for extremely resource-limited environments (e.g. embedded systems like a network 
+switch). If your application is not highly resource constrained, LZ4 is a better choice.
+
+The native implementation of Snappy is provided by the `SnappyNativeCompressor` and `SnappyNativeDecompressor`
+classes. The Java implementation is provided by the `SnappyJavaCompressor` and `SnappyJavaDecompressor`
+
+The Snappy framed format is supported by the `SnappyFramedInputStream` and `SnappyFramedOutputStream`
+classes.
+
+## [LZO](https://www.oberhumer.com/opensource/lzo/) 
+LZO is only provided for compatibility with existing systems that use LZO. We recommend 
+reencoding LZO data using Zstandard or LZ4. 
+
+The Java implementation of LZO is provided by the `LzoJavaCompressor` and `LzoJavaDecompressor` classes.
+Due to licensing issues, the LZO only has a Java implementation which is based on the LZ4 codebase.
+
+## Deflate
+Deflate is the block compression algorithm used by the `gzip` and `zlib` libraries. Deflate is
+provided for compatibility with existing systems that use Deflate. We recommend reencoding
+Deflate data using Zstandard which provides superior compression and performance.
+
+The implementation of Deflate is provided by the `DeflateCompressor` and `DeflateDecompressor` classes.
+This is implemented in the built-in Java libraries which internally use the native code.
+
+# Hadoop Compression
 
 In addition to the raw block encoders, there are implementations of the
-Hadoop CompressionCodec (Streaming) for each algorithm. They are
-typically 300% faster than the JNI wrappers.
+Hadoop Streams for the above algorithms. In addition, implementations of
+GZIP and BZip2 are provided so that all standard Hadoop algorithms are available.
+
+The `HadoopStreams` class provides a factory for creating `InputStream` and `OutputStream`
+implementations without the need for any Hadoop dependencies.  For environments 
+that have Hadoop dependencies, each algorithm also provides a `CompressionCodec` class.
 
 # Requirements
 
-This library requires a Java 1.8+ virtual machine containing the `sun.misc.Unsafe` interface running on a little endian platform.
+This library requires a Java 22+ virtual machine containing the `sun.misc.Unsafe` interface running on a little endian platform.
```

Comment:
Would you consider building a multi-release JAR, so us miserable folks stuck on Java 1.8 can still use the pure Java bits of this library?